### ISSUE TITLE
[web] Support forced-colors feature

### ISSUE THREAD

**Issue by nbayati** [flutter/flutter#118853]
Support Windows' forced-colors feature in web.

related to: https://github.com/flutter/flutter/issues/75883

**Comment by chunhtai** [comment]
cc @QuncCccccc

**Comment by yjbanov** [comment]
Self-assigned to do the web engine part.

**Comment by yjbanov** [comment]
Optimistically assigning @QuncCccccc for the framework part.

**Comment by yjbanov** [comment]
With https://github.com/flutter/flutter/pull/163335 apps that use custom themes should be able to wire up the colors into their themes. The framework part of this item will only wire up the colors into Flutter's standard themes.
